src/function/table/information_schema/information_schema_schemata.cpp

```diff
@@ -0,0 +1,106 @@
+#include "duckdb/common/exception.hpp"
+#include "duckdb/main/client_context.hpp"
+#include "duckdb/transaction/transaction.hpp"
+
+using namespace std;
+
+namespace duckdb {
+
+struct InformationSchemaSchemataData : public FunctionOperatorData {
+	InformationSchemaSchemataData() : offset(0) {
+	}
+
+	vector<SchemaCatalogEntry *> entries;
+	idx_t offset;
+};
+
+static unique_ptr<FunctionData> information_schema_schemata_bind(ClientContext &context, vector<Value> &inputs,
+                                                                 unordered_map<string, Value> &named_parameters,
+                                                                 vector<LogicalType> &return_types,
+                                                                 vector<string> &names) {
+	names.push_back("catalog_name");
+	return_types.push_back(LogicalType::VARCHAR);
+
+	names.push_back("schema_name");
+	return_types.push_back(LogicalType::VARCHAR);
+
+	names.push_back("schema_owner");
+	return_types.push_back(LogicalType::VARCHAR);
+
+	names.push_back("default_character_set_catalog");
+	return_types.push_back(LogicalType::VARCHAR);
+
+	names.push_back("default_character_set_schema");
+	return_types.push_back(LogicalType::VARCHAR);
+
+	names.push_back("default_character_set_name");
+	return_types.push_back(LogicalType::VARCHAR);
+
+	names.push_back("sql_path");
+	return_types.push_back(LogicalType::VARCHAR);
+
+	return nullptr;
+}
+
+unique_ptr<FunctionOperatorData>
+information_schema_schemata_init(ClientContext &context, const FunctionData *bind_data, ParallelState *state,
+                                 vector<column_t> &column_ids,
+                                 unordered_map<idx_t, vector<TableFilter>> &table_filters) {
+	auto result = make_unique<InformationSchemaSchemataData>();
+
+	// scan all the schemas and collect them
+	auto &transaction = Transaction::GetTransaction(context);
+	Catalog::GetCatalog(context).schemas->Scan(
+	    transaction, [&](CatalogEntry *entry) { result->entries.push_back((SchemaCatalogEntry *)entry); });
+	// get the temp schema as well
+	result->entries.push_back(context.temporary_objects.get());
+
+	return move(result);
+}
+
+void information_schema_schemata(ClientContext &context, const FunctionData *bind_data,
+                                 FunctionOperatorData *operator_state, DataChunk &output) {
+	auto &data = (InformationSchemaSchemataData &)*operator_state;
+	if (data.offset >= data.entries.size()) {
+		// finished returning values
+		return;
+	}
+	idx_t next = min(data.offset + STANDARD_VECTOR_SIZE, (idx_t)data.entries.size());
+	output.SetCardinality(next - data.offset);
+
+	// start returning values
+	// either fill up the chunk or return all the remaining columns
+	for (idx_t i = data.offset; i < next; i++) {
+		auto index = i - data.offset;
+		auto &entry = data.entries[i];
+
+		// return values:
+		// "catalog_name", PhysicalType::VARCHAR
+		// TODO(jwills): how to determine this?
```

Comment:
are you looking for the catalog name? its in entry->name no?

Reply:
Ah no there is only one catalog indeed. So "main" is fine, NULL would also be an option. Depends on how tools handle this I think.

Reply:
In postgres it is ["always the current database"](https://www.postgresql.org/docs/9.1/infoschema-schemata.html), which I think is equivalent to "main" from a DuckDB perspective?

Reply:
"main" is the name of the primary schema, I think setting this to NULL makes more sense. Either that, or setting it to the current database file, as is the case in `pragma_database_list`.

Reply:
@Mytherin my read from [pragma_database_list](https://github.com/cwida/duckdb/blob/master/test/sql/pragma/test_pragma_database_list.test) was that the name of the database is always "main".

For my use case, the value of this field doesn't actually matter, so if the consensus is for NULL, I am 👍 to make that change.